### PR TITLE
ES6 migration: autoFocuser, dom, scrollManager

### PR DESCRIPTION
**Pull request by dmitrylyzo**
**Changes**
* Migrate ScrollManger to ES6
* Migrate AutoFocuser to ES6
* Migrate DOM-module to ES6
* Add default export for compatibility `import` <-> `require`

**Issues**
#868

### CODE REVIEW

File: src/components/autoFocuser.js [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 
     /**
      * Previously selected element.
      */
-    var activeElement;
+    let activeElement;
 
     /**
-     * Returns true if AutoFocuser is enabled.
+     * Returns _true_ if AutoFocuser is enabled.
```
Comment (dkanada) [inline src/components/autoFocuser.js]: I don't think we want to wrap this in underscores.

File: src/components/autoFocuser.js [hunk 2]
```diff
@@ -28,24 +35,19 @@ define(["focusManager", "layoutManager"], function (focusManager, layoutManager)
...
-    /**
-     * Create an array from some source.
-     */
-    var arrayFrom = Array.prototype.from || function (src) {
-        return Array.prototype.slice.call(src);
-    }
-
     /**
      * Set focus on a suitable element, taking into account the previously selected.
+     * @param {HTMLElement} [container] - element to limit scope
```
Comment (dkanada) [inline src/components/autoFocuser.js]: I also don't think we were wrapping parameters in brackets.

File: src/components/scrollManager.js [hunk 3]
```diff
@@ -433,39 +443,39 @@ define(["dom", "browser", "layoutManager"], function (dom, browser, layoutManage
     /**
      * Scrolls the document to a given element.
      *
-     * @param {HTMLElement} element target element of scroll task
-     * @param {boolean} [smooth=false] smooth scrolling
+     * @param {HTMLElement} element - target element of scroll task
+     * @param {boolean} [smooth=false] - smooth scrolling
```
Comment (dkanada) [inline src/components/scrollManager.js]: Will this value always be false?

File: src/components/dom.js [hunk 4]
```diff
@@ -104,46 +158,60 @@ define([], function () {
...
-    var _animationEvent;
-    function whichAnimationEvent() {
+    /**
+     * Name of animation end event.
+     */
+    let _animationEvent;
 
+    /**
+     * Returns name of animation end event.
+     * @returns {string} Name of animation end event
```
Comment (dkanada) [inline src/components/dom.js]: @MrTimscampi period or no period?

File: src/components/autoFocuser.js (hunk 1 above)
Comment (dmitrylyzo) [reply src/components/autoFocuser.js]: if you turn on `plugins/markdown`, this makes _italic_

File: src/components/autoFocuser.js (hunk 2 above)
Comment (dmitrylyzo) [reply src/components/autoFocuser.js]: this marks optional argument

File: src/components/scrollManager.js (hunk 3 above)
Comment (dmitrylyzo) [reply src/components/scrollManager.js]: this marks optional + default is _false_

File: src/components/dom.js (hunk 4 above)
Comment (dmitrylyzo) [reply src/components/dom.js]: Currently it looks like this
doc1

Comment (MrTimscampi) [reply src/components/dom.js]: I'd add a period (that's what I've been doing on every JSDoc comment).

I feel like we should be writing these both with in-source legibility (hence my use of "-" to separate parameter names and descriptions) and periods (to look nicer in the compiled docs).

Eventually, we probably want the JSDoc to be available in the developer doc to help people understand the web client before diving in.

Comment (dmitrylyzo) [reply src/components/dom.js]: doc2

Comment (MrTimscampi) [reply src/components/dom.js]: As discussed on Matrix with @dkanada, adopting the Google Style Guide for JSDoc is probably a good idea: https://google.github.io/styleguide/jsguide.html#jsdoc

We're nearly there anyway, so...

We'll make an addition to the developer documentation about it at some point.

Comment (dkanada) [reply src/components/dom.js]: Yeah, we can ignore this change for now and then fix all issues with the linting.

**Pull request merged**